bin/generate.py

```diff
@@ -1291,44 +1305,39 @@ def count(yaml):
                 if isinstance(elem, dict):
                     for key in elem:
                         if is_testcase(elem[key]) and numbered:
-                            num_numbered_testcases += 1
+                            num_numbered_testcases += parse_count(elem[key])
                         elif is_directory(elem[key]):
                             count(elem[key])
 
         count(yaml)
 
         # Main recursive parsing function.
         # key: the yaml key e.g. 'testcase'
-        # name: the possibly numbered name e.g. '01-testcase'
+        # name_gen: each call should result in the next (possibly numbered) name e.g. '01-testcase'
         # Returns either a single Rule or a list of Rules
-        def parse(key, name, yaml, parent):
-            nonlocal testcase_id
-
+        def parse(key, name_gen, yaml, parent):
+            name = name_gen()
             check_type('Testcase/directory', yaml, [type(None), str, dict], parent.path)
             if not is_testcase(yaml) and not is_directory(yaml):
                 fatal(f'Could not parse {parent.path / name} as a testcase or directory.')
 
             if is_testcase(yaml):
-                count = 1
-                if yaml is not None and 'count' in yaml and isinstance(yaml['count'], int):
-                    count = yaml['count']
-                if count > 100:
-                    warn(f'Found count: {count}, limited to 100.')
-                    count = 100
+                count = parse_count(yaml, True)
 
                 if not config.COMPILED_FILE_NAME_REGEX.fullmatch(name + '.in'):
                     error(f'Testcase \'{parent.path}/{name}.in\' has an invalid name.')
                     return None
 
                 ts = []
                 for count_index in range(count):
-                    curname = name if count == 1 else name + f'-{count_index+1:0{len(str(count))}}'
+                    if count_index > 0:
+                        name = name_gen()
 
                     # If a list of testcases was passed and this one is not in it, skip it.
-                    if not self.process_testcase(parent.path / curname):
+                    if not self.process_testcase(parent.path / name):
                         continue
```

Comment:
I have the feeling that this does not really have a proper semantics for test cases with a `count:` :thinking:

Reply:
testcases with count behave the same way as any other... i.e. have regular names?

Reply:
I never actually used `bt generate [testcases...]`, so I'm also not 100% sure what the syntax should be for the testcases that should be generated. Any syntax that I try, results in `bt generate` simply printing "Done" twice, with nothing happening, even though I make changes to the relevant test cases in `generators.yaml` :thinking:

Reply:
you should give it a path like `data/secret/blub.in` i guess?